examples/DataApi-NFT/template/modules/shared_data/src/main/scala/com/my/nft_example/shared_data/calculated_state/CalculatedState.scala

```diff
@@ -0,0 +1,49 @@
+package com.my.nft_example.shared_data.calculated_state
+
+import cats.effect.IO
+import com.my.nft_example.shared_data.types.Types.NFTUpdatesCalculatedState
+import eu.timepit.refined.types.numeric.NonNegLong
+import io.circe.syntax.EncoderOps
+import org.tessellation.schema.SnapshotOrdinal
+import org.tessellation.security.hash.Hash
+
+import java.security.MessageDigest
+
+object CalculatedState {
+
+  private var maybeNFTUpdatesCalculatedState: (SnapshotOrdinal, NFTUpdatesCalculatedState) = (
```

Comment:
@AlexBrandes this is the internal variable that I mentioned yesterday. The patch Krystian is working on doesn't apply in this case because there needs to be a place to store the `NFTCalculatedState`. This would likely need to move to seem on-disk persistence solution but perhaps this has been a discussion with @ryle-goehausen and the wider protocol team already.

Reply:
As it is this is definitely not safe and has the potential for race conditions on other threads. Yesterday I showed @IPadawans how to wrap it in an `AtomicReference` but now that the _dataApplication_ hook has been updated we can use the `make` pattern to create our services and use a `Ref`.